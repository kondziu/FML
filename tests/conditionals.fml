// Expected output:
// > true things are true
// > false things are not true
// > null is not true
// > zero is true
// > one is true
// > x is true
// > y is not true
// > true-false block is not true
// > false-true block is true
// > z is not true
// > now z is true
// > consequent is true
// > alternative is false
// > consequent is always true
// > alternative is sometimes null

if true
then print("true things are true\n")
else print("true things are not true\n");

if false
then print("false things are true\n")
else print("false things are not true\n");

if null
then print("null is true\n")
else print("null is not true\n");

if 0
then print("zero is true\n")
else print("zero is not true\n");

if 1
then print("one is true\n")
else print("one is not true\n");

let x = true;
if x
then print("x is true\n")
else print("x is not true\n");

let y = false;
if y
then print("y is true\n")
else print("y is not true\n");

if begin true; false; end
then print("true-false block is true\n")
else print("true-false block is not true\n");

if begin false; true end
then print("false-true block is true\n")
else print("false-true block is not true\n");

let z = false;
z <- if z
then print("z is true\n")
else begin
  print("z is not true\n");
  true;
end;

if z
then print("now z is true\n")
else print("now z is not true\n");

let consequent = if true then true else false;
print("consequent is ~\n", consequent);

let alternative = if false then true else false;
print("alternative is ~\n", alternative);

consequent <- if true then true;
print("consequent is always ~\n", consequent);

alternative <- if false then true;
print("alternative is sometimes ~\n", alternative);

