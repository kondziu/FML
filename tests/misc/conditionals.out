true things are true
false things are not true
null is not true
zero is true
one is true
x is true
y is not true
true-false block is not true
false-true block is true
z is not true
now z is true
consequent is true
alternative is false
consequent is always true
alternative is sometimes null
